x-pack/plugins/aiops/public/components/log_rate_analysis/log_rate_analysis_results.tsx

```diff
@@ -221,14 +224,18 @@ export const LogRateAnalysisResults: FC<LogRateAnalysisResultsProps> = ({
         ((Array.isArray(remainingFieldCandidates) && remainingFieldCandidates.length > 0) ||
           groupsMissing)
       ) {
-        setOverrides({ loaded, remainingFieldCandidates, significantTerms: data.significantTerms });
+        setOverrides({
+          loaded,
+          remainingFieldCandidates,
+          significantItems: data.significantItems as AiopsLogRateAnalysisSchemaSignificantItem[],
```

Comment:
Looks like there is a lot of type casting needed with 'as' - is there a way to remove the need for that?

Reply:
I looked into this and I think there is no way around it in this case, all the examples I found for conditional types do the same (or fall back to `any`). There is an issue about this in the TypeScript repro which describes this problem, the issue was closed as a design limitation: https://github.com/microsoft/TypeScript/issues/22735